app/src/main/java/com/vultisig/wallet/ui/screens/vault_settings/VaultSettingsViewModel.kt

```diff
@@ -40,9 +58,30 @@ internal open class VaultSettingsViewModel @Inject constructor(
     fun onEvent(event: VaultSettingsEvent) {
         when (event) {
             Backup -> backupVault()
-            Delete -> deleteVault()
+            ShowConfirmDeleteScreen -> showConfirmDeleteDialog()
+            DismissConfirmDeleteScreen -> dismissConfirmDeleteDialog()
+            Delete -> delete()
             ErrorDownloadFile -> errorBackUp()
             is SuccessBackup -> successBackup(event.fileName)
+            is ChangeCheckCaution -> changeCheckCaution(event.index, event.isChecked)
+        }
+    }
+
+    private fun dismissConfirmDeleteDialog() {
+        _uiModel.update {
+            it.copy(showDeleteConfirmScreen = false)
+        }
+    }
+
+    private fun changeCheckCaution(index: Int, checked: Boolean) {
+        _uiModel.update {
+            val checkedCautionIndexes = it.checkedCautionIndexes.toMutableList()
+            if (checked) checkedCautionIndexes.add(index)
+            else checkedCautionIndexes.remove(index)
+            it.copy(
+                checkedCautionIndexes = checkedCautionIndexes,
+                isDeleteButtonEnabled = checkedCautionIndexes.size == it.cautionsBeforeDelete.size
+            )
         }
```

Comment:
The lambda function in the update method can be evaluated several times. Consider removing all calculations from lambda.

Reply:
Done